### PR TITLE
Clean up log messages around faucet request.

### LINKED ISSUE #650: decrease log severity for 'Failed to request Eth from Faucet'
All attempts that will be retried should be WARN (with "Retrying" in the msg), and the last (6th) one should be ERROR.

### PR DESCRIPTION
**Pull request by maaktweluit**
Resolves #650 

```
[2020-10-07T11:40:59Z INFO  ya_gnt_driver::gnt::faucet] Requesting Eth from faucet
[2020-10-07T11:41:00Z WARN  ya_gnt_driver::gnt::faucet] Failed to request Eth from Faucet, retrying (1/6): LibraryError("Failed to connect to host: Timeout out while establishing connection")
[2020-10-07T11:41:03Z WARN  ya_gnt_driver::gnt::faucet] Failed to request Eth from Faucet, retrying (2/6): LibraryError("Failed to connect to host: Timeout out while establishing connection")
[2020-10-07T11:41:06Z WARN  ya_gnt_driver::gnt::faucet] Failed to request Eth from Faucet, retrying (3/6): LibraryError("Failed to connect to host: Timeout out while establishing connection")
[2020-10-07T11:41:09Z WARN  ya_gnt_driver::gnt::faucet] Failed to request Eth from Faucet, retrying (4/6): LibraryError("Failed to connect to host: Timeout out while establishing connection")
[2020-10-07T11:41:12Z WARN  ya_gnt_driver::gnt::faucet] Failed to request Eth from Faucet, retrying (5/6): LibraryError("Failed to connect to host: Timeout out while establishing connection")
[2020-10-07T11:41:15Z ERROR ya_gnt_driver::gnt::faucet] Failed to request Eth from Faucet, tried 6 times.: LibraryError("Failed to connect to host: Timeout out while establishing connection")
```

### CODE REVIEW

**Review by Wiezzel — APPROVED** [review]

File: core/payment-driver/gnt/src/gnt/faucet.rs [hunk 1]
```diff
@@ -48,11 +48,12 @@ impl EthFaucetConfig {
     }
 
     pub async fn request_eth(&self, address: Address) -> GNTDriverResult<()> {
-        log::debug!("request eth");
+        log::info!("Requesting Eth from faucet");
```
Comment (tworec) [inline core/payment-driver/gnt/src/gnt/faucet.rs:51]: 👍

File: core/payment-driver/gnt/src/gnt/faucet.rs [hunk 2]
```diff
@@ -62,20 +63,36 @@ impl EthFaucetConfig {
...
+                    log::warn!(
+                        "Failed to request Eth from Faucet, retrying ({}/{}): {:?}",
+                        i + 1,
+                        MAX_ETH_FAUCET_REQUESTS,
+                        e
+                    );
+                    tokio::time::delay_for(ETH_FAUCET_SLEEP).await;
+                }
             } else {
+                log::info!("Succesfully requested Eth, waiting...");
```
Comment (tworec) [inline core/payment-driver/gnt/src/gnt/faucet.rs:93]: 👍

File: core/payment-driver/gnt/src/gnt/faucet.rs [hunk 3]
```diff
@@ -62,20 +63,36 @@ impl EthFaucetConfig {
...
                 tokio::time::delay_for(INIT_ETH_SLEEP).await;
+                log::info!("Finished requesting Eth from faucet");
```
Comment (tworec) [inline core/payment-driver/gnt/src/gnt/faucet.rs:95]: do we need separate logs for it? I believe sleep basically can't fail... Is this a valuable info that we sleeped well :) ?

File: core/payment-driver/gnt/src/gnt/faucet.rs [hunk 4]
```diff
@@ -62,20 +63,36 @@ impl EthFaucetConfig {
...
+                // Do not warn nor sleep at the last try.
+                if i >= MAX_ETH_FAUCET_REQUESTS - 1 {
+                    log::error!(
+                        "Failed to request Eth from Faucet, tried {} times.: {:?}",
+                        MAX_ETH_FAUCET_REQUESTS,
+                        e
+                    );
+                } else {
+                    log::warn!(
+                        "Failed to request Eth from Faucet, retrying ({}/{}): {:?}",
```
Comment (tworec) [inline core/payment-driver/gnt/src/gnt/faucet.rs:85]:
```suggestion
                        "Retrying ({}/{}) to request Eth from Faucet after failure: {:?}",
```

File: core/payment-driver/gnt/src/gnt/faucet.rs [hunk 5]
```diff
@@ -62,20 +63,36 @@ impl EthFaucetConfig {
...
 
             Err(GNTDriverError::LibraryError(resp.into_owned()))
         }
 
-        for _ in 0..MAX_ETH_FAUCET_REQUESTS {
+        for i in 0..MAX_ETH_FAUCET_REQUESTS {
             if let Err(e) = try_request_eth(&client, &request_url).await {
-                log::error!("Failed to request Eth from Faucet: {:?}", e);
-                tokio::time::delay_for(ETH_FAUCET_SLEEP).await;
+                // Do not warn nor sleep at the last try.
```
Comment (tworec) [inline core/payment-driver/gnt/src/gnt/faucet.rs:76]: 👍 comments are goood! :)

File: core/payment-driver/gnt/src/gnt/faucet.rs [hunk 6]
```diff
@@ -62,20 +63,36 @@ impl EthFaucetConfig {
                 .await
                 .map_err(|e| GNTDriverError::LibraryError(e.to_string()))?;
             let resp = std::string::String::from_utf8_lossy(body.as_ref());
+            log::debug!("resp: {}", resp);
```
Comment (tworec) [inline core/payment-driver/gnt/src/gnt/faucet.rs:66]:
```suggestion
            log::debug!("raw faucet response: {}", resp);
```

File: core/payment-driver/gnt/src/gnt/faucet.rs [hunk 7]
```diff
@@ -48,11 +48,12 @@ impl EthFaucetConfig {
     }
 
     pub async fn request_eth(&self, address: Address) -> GNTDriverResult<()> {
-        log::debug!("request eth");
+        log::info!("Requesting Eth from faucet");
         let client = awc::Client::new();
         let request_url = format!("{}/{}", &self.faucet_address, utils::addr_to_str(address));
 
         async fn try_request_eth(client: &awc::Client, url: &str) -> GNTDriverResult<()> {
+            log::debug!("url: {}", url);
```
Comment (tworec) [inline core/payment-driver/gnt/src/gnt/faucet.rs:56]: I believe we do not need to repeat this 6 times.. I'd place it right after `info!` log above (line 51)

**Review by tworec — APPROVED** [review]
lgtm. just nitpicks

File: core/payment-driver/gnt/src/gnt/faucet.rs (hunk 3 above)
Comment (maaktweluit) [reply core/payment-driver/gnt/src/gnt/faucet.rs:95]: `waiting...` as last line seemed like too much of an open end.

you are correct the sleep can not go wrong

File: core/payment-driver/gnt/src/gnt/faucet.rs (hunk 7 above)
Comment (maaktweluit) [reply core/payment-driver/gnt/src/gnt/faucet.rs]: good one! updated

**Comment by maaktweluit** [comment]
@tworec @Wiezzel Thanks for the reviews and input!

Fixed all comments, replied to the "Completed" log message.
Please let me know if we should keep or remove this line and we are ready to merge :)

**Comment by etam** [comment]
:metal: :smiling_imp: :metal:

**Comment by tworec** [comment]
LGTM. go and merge it!

**Pull request merged**